core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -254,16 +256,16 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
 
             case Left(Errors.COORDINATOR_LOAD_IN_PROGRESS) =>
               info(s"I am loading the transaction partition that contains $transactionalId while my current coordinator epoch is $coordinatorEpoch; " +
-                s"so appending $newMetadata to transaction log since the loading process will continue the left work")
+                s"so cancel appending $newMetadata to transaction log since the loading process will continue the left work")
 
             case Right(Some(epochAndMetadata)) =>
               if (epochAndMetadata.coordinatorEpoch == coordinatorEpoch) {
-                debug(s"Updating $transactionalId's transaction state to $txnMetadata with coordinator epoch $coordinatorEpoch for $transactionalId succeeded")
+                debug(s"Sending $transactionalId's transaction markers for $txnMetadata with coordinator epoch $coordinatorEpoch for $transactionalId succeeded, trying to append complete transaction log now")
 
                 tryAppendToLog(TxnLogAppend(transactionalId, coordinatorEpoch, txnMetadata, newMetadata))
               } else {
-                info(s"Updating $transactionalId's transaction state to $txnMetadata with coordinator epoch $coordinatorEpoch for $transactionalId failed after the transaction markers " +
-                  s"has been sent to brokers. The cached metadata have been changed to $epochAndMetadata since preparing to send markers")
+                info(s"The cached metadata for $transactionalId have been changed to $epochAndMetadata after completed sending the markers while its old epoch " +
```

Comment:
nit: this reads a bit awkwardly: "after completed sending the markers while its old epoch was $coordinatorEpoch when preparing to send marker." Maybe we could write what happened and include additional metadata in parenthesis at the end?